### PR TITLE
Add technologies contextual analysis support flag

### PR DESCRIPTION
**Pull request by omerzi**
- [x] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

**Review by sverdlov93 — APPROVED** [review]

**Review by sverdlov93 — COMMENTED** [review]
see my inline comments

File: xray/audit/jas/applicabilitymanager.go [hunk 1]
```diff
@@ -119,78 +91,57 @@ func newApplicabilityScanManager(xrayScanResults []services.ScanResponse, depend
...
-
-// This function gets a liat of xray scan responses that contains direct and indirect violations, and returns only direct
-// violation of the scanned project, ignoring indirect violations
-func extractXrayDirectViolations(xrayScanResults []services.ScanResponse, directDependencies []string) []services.Violation {
-	xrayViolationsDirectDependency := []services.Violation{}
-	for _, violation := range getXrayViolations(xrayScanResults) {
-		for _, dep := range directDependencies {
-			if _, ok := violation.Components[dep]; ok {
-				xrayViolationsDirectDependency = append(xrayViolationsDirectDependency, violation)
+// This function gets a liat of xray scan responses that contains direct and indirect vulnerabilities, and returns only direct
```
[the remark this answers is not in the document]
Comment (omerzi) [reply xray/audit/jas/applicabilitymanager.go:94]: done

File: xray/audit/jas/applicabilitymanager.go [hunk 2]
```diff
@@ -202,7 +153,15 @@ func (a *ApplicabilityScanManager) run() (err error) {
 }
 
 func (a *ApplicabilityScanManager) directDependenciesExist() bool {
-	return len(createCveList(a.xrayVulnerabilities, a.xrayViolations)) > 0
+	return a.directDependenciesCves.Size() > 0
+}
+
+func (a *ApplicabilityScanManager) eligibleForApplicabilityScan(technologies []coreutils.Technology) bool {
+	if !a.directDependenciesExist() || !coreutils.ContainsApplicabilityScannableTech(technologies) {
```
[the remark this answers is not in the document]
Comment (omerzi) [reply xray/audit/jas/applicabilitymanager.go:160]: done

File: xray/audit/jas/applicabilitymanager.go [hunk 3]
```diff
@@ -50,8 +47,7 @@ func getApplicabilityScanResults(results []services.ScanResponse, dependencyTree
 			err = errors.Join(err, cleanupFunc())
 		}
 	}()
-	if !applicabilityScanManager.eligibleForApplicabilityScan() {
-		log.Debug("The conditions for running the applicability scan are not met. Skipping...")
+	if !applicabilityScanManager.eligibleForApplicabilityScan(scannedTechnologies) {
```
[the remark this answers is not in the document]
Comment (omerzi) [reply xray/audit/jas/applicabilitymanager.go:50]: done

**Pull request merged**